modin/pandas/test/dataframe/test_default.py

```diff
@@ -132,7 +132,12 @@ def test_to_numpy(data):
 @pytest.mark.parametrize("data", test_data_values, ids=test_data_keys)
 def test_partition_to_numpy(data):
     frame = pd.DataFrame(data)
-    for partition in frame._query_compiler._modin_frame._partitions.flatten().tolist():
+    qc = frame._query_compiler
+    if get_current_execution() == "Client":
+        modin_frame = qc._service._qc[qc._id]._modin_frame
+    else:
+        modin_frame = qc._modin_frame
+    for partition in modin_frame._partitions.flatten().tolist():
```

Comment:
Is it expected that the client query compiler would have partitions?

Reply:
The client query compiler won't have partitions, but here we peak deep into its internals to get partitions-- specifically we get partitions from the `BaseQueryCompiler` in the container in the client query compiler.